File: angularclient/src/app/post.ts
import { Comment } from '../app/comment';
export class Post {

    id: number;
    title: String;
    content: String;
    comments: Comment[];

    getId() {
        return this.id;
    }

    getTitle() {
        return this.title;
    }

    getContent() {
        return this.content;
    }

    getComments() {
        return this.comments;
    }

    setId(id: number){
         this.id = id;
    }

    setTitle(title: String) {
        this.title = title;
    }

    setContent(content: String) {
        this.content = content;
    }

    setComments(comments: Comment[]) {
        this.comments = comments;
    }

}

File: src/main/java/com/manifestcorp/Blog/model/Post.java
package com.manifestcorp.Blog.model;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonProperty;

@Entity(name = "Post")
@Table(name = "post")

public class Post extends BaseEntity {
	
	private String title;
	
	//The One to many here is referring to the list of comments below.
	@OneToMany(
		mappedBy = "post",
		cascade = CascadeType.ALL,
		orphanRemoval = true
			)
	
	private List<Comment> comments = new ArrayList<Comment>();
	
	private String content;

	
	public Post() {}
	
	public Post( @JsonProperty("title") String title, @JsonProperty("content") String content) {
		this.title = title;
		this.content = content;
	}
	
	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public List<Comment> getComments() {
		return comments;
	}

	public void setComments(List<Comment> comments) {
		this.comments = comments;
	}

	public String getContent() {
		return content;
	}

	public void setContent(String content) {
		this.content = content;
	}
	
	public void addComment(Comment comment) {
		comments.add(comment);
		comment.setPost(this);
	}
	
}
